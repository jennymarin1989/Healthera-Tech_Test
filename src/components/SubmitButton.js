import React from 'react'
import PortalButton from './styling/portalButton';

const SubmitButton = props => {

    return(
        <PortalButton id={props.id} onClick={props.handlePortalClicked}>{props.name}</PortalButton>
    );
};

export default SubmitButton;