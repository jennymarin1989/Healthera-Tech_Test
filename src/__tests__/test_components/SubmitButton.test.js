import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SubmitButton from '../../components/SubmitButton';

describe("SubmitButton", () => {
	
	let submit = shallow(<SubmitButton />);

	it("renders correctly", () => {
		expect(submit).toMatchSnapshot();
	});

	describe('When a user clicks a button', () => {
		it('should call the handleDataSumittedButton function', ()=>{
			let mockhandlePortalClicked = jest.fn()
			let submit = shallow(<SubmitButton handlePortalClicked = {mockhandlePortalClicked}/>);
			submit.find('#submitButton').simulate('click', 'using prototype');
			expect(mockhandlePortalClicked).toBeCalled();
		})
	});

	describe("renders component correctly", () => {
		it("renders PortalButton component correctly", () => {
			let patientButton = renderer
			.create(<button>Patient Portal</button>)
      		.toJSON();
   			 expect(patientButton).toMatchSnapshot();
		})
	})
});