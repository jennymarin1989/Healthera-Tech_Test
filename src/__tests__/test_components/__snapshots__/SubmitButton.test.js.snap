// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders component correctly renders PortalButton component correctly 1`] = `
<button
  className="sc-bdVaJa jCyAds"
>
  Patient Portal
</button>
`;

exports[`SubmitButton renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": undefined,
      "id": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "id": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
